### PR TITLE
Added door related events

### PR DESCRIPTION
**Pull request by nihal111**
New door related events that make the door inside Core module easily extendable and multiplayer ready.

### CODE REVIEW

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: modules/Core/src/main/java/org/terasology/core/logic/door/DoorSystem.java [hunk 1]
```diff
@@ -182,18 +183,39 @@ private boolean canAttachTo(Vector3i doorPos, Side side) {
     @ReceiveEvent(components = {DoorComponent.class, BlockRegionComponent.class, LocationComponent.class})
     public void onFrob(ActivateEvent event, EntityRef entity) {
         DoorComponent door = entity.getComponent(DoorComponent.class);
-        Side newSide = (door.isOpen) ? door.closedSide : door.openSide;
+        if (door.isOpen) {
```
Comment (SkySom) [inline modules/Core/src/main/java/org/terasology/core/logic/door/DoorSystem.java]: As mentioned in Slack, you send the CloseDoorEvent in both logic branches, I assume one should be open?

File: modules/Core/src/main/java/org/terasology/core/logic/door/DoorSystem.java [hunk 2]
```diff
@@ -182,18 +183,39 @@ private boolean canAttachTo(Vector3i doorPos, Side side) {
...
+
+    @ReceiveEvent(components = {DoorComponent.class, BlockRegionComponent.class, LocationComponent.class})
+    public void openDoor(OpenDoorEvent event, EntityRef entity, DoorComponent door) {
+        Side newSide = door.openSide;
+        BlockRegionComponent regionComp = entity.getComponent(BlockRegionComponent.class);
+        Block bottomBlock = door.bottomBlockFamily.getBlockForPlacement(worldProvider, blockEntityRegistry, regionComp.region.min(), newSide, Side.TOP);
+        worldProvider.setBlock(regionComp.region.min(), bottomBlock);
+        Block topBlock = door.topBlockFamily.getBlockForPlacement(worldProvider, blockEntityRegistry, regionComp.region.max(), newSide, Side.TOP);
+        worldProvider.setBlock(regionComp.region.max(), topBlock);
+        if (door.openSide != null) {
```
Comment (SkySom) [inline modules/Core/src/main/java/org/terasology/core/logic/door/DoorSystem.java]: I assume you mean door.openSound matching what you for close sound?

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by Cervator** [comment]
Tested out locally - merged! :-)

There is still a quirk where the password doors in AdventureAssets only work for the player who placed it but they *do* work. So that's probably just some missing tweak we can find a little later.

**Pull request merged**